miden-lib/asm/sat/prologue.masm

```diff
@@ -0,0 +1,363 @@
+proc.process_acct_data
+    # compute the address to store the account data
+    push.ACCT_DATA_PTR.1 add
+
+    # prepare the stack for absorbing account data
+    padw padw padw
+
+    # absorb the account data
+    adv_pipe adv_pipe
+
+    # extract digest from hasher rate elements (h_0, ..., h_3)
+    dropw swapw dropw
+
+    # assert that the account hash matches the hash provided by the stack public input
+    padw push.ACCT_HASH_PTR mem_loadw assert_eqw
+
+    # assert the account id matches the account id provided via stack public inputs
+    push.ACCT_ID_PTR mem_load
+    push.ACCT_DATA_PTR.1 add mem_load assert_eq
+
+    # clear the stack
+    drop
+end
+
+# CONSUMED NOTES DATA
+# =================================================================================================
+
+#! Computes a pointer to the memory address at which the data associated with a note with index i
+#! is stored.
+#!
+#! Stack: [i]
+#! Output: [ptr]
+proc.compute_note_data_ptr
+    # compute the address to store the consumed note data
+    push.1 add push.1024 mul push.CONSUMED_NOTE_DATA_OFFSET add
+end
+
+#! Computes a pointer to the memory address at which the nullifier associated a note with index i
+#! is stored.
+#!
+#! Stack: [i]
+#! Output: [ptr]
+proc.compute_nullifier_ptr
+    # compute the address to store the nullifier
+    push.CONSUMED_NOTE_DATA_OFFSET.1 add add
+end
+
+#! Ingests data for consumed note i from the advice provider and stores it in memory at the
+#! deterministic memory address. This includes computing and storing the nullifier and the
+#! note hash.
+#!
+#! Stack: [i]
+#! Advice Tape: [cn1_sn3, cn1_sn2, cn1_sr1, cn1_sr0,
+#!               cn1_sr3, cn1_sr2, cn1_sr1, cn1_sr0,
+#!               cn1_ir3, cn1_ir2, cn_ir1, cn_ir0,
+#!               cn1_vr3, cn1_vr2, cn1_vr1, cn1_vr0,
+#!               cn1_na,
+#!               cn1_a1_3, cn1_a1_2, cn1_a1_1, cn1_a1_0,
+#!               cn1_a2_3, cn1_a2_2, cn1_a2_1, cn1_a2_0]
+#! Output: []
+#!
+#! - cn1_sn is the serial number of consumed note 1.
+#! - cn1_sr is the script root of consumed note 1.
+#! - cn1_ir is the inputs root of consumed note 1.
+#! - cn1_vr is the vault root of consumed note 1.
+#! - cn1_na is the number of assets in consumed note 1.
+#! - cn1_a1 is the first asset of consumed note 1.
+#! - cn1_a2 is the second asset of consumed note 1.
+proc.process_consumed_note
+    # ingest core note data
+    # ---------------------------------------------------------------------------------------------
+
+    # compute address to store note hash
+    exec.compute_note_data_ptr
+
+    # compute address to store raw note data
+    dup push.1 add
+
+    # absorb note data from the advice provider
+    padw padw padw
+    adv_pipe adv_pipe
+
+    # extract nullifier
+    dropw swapw dropw
+
+    # compute address for nullifier
+    # dup.6 = counter / note index
+    dup.6 exec.compute_nullifier_ptr
+
+
+    # todo assert nullifier does not exist in nullifier db
```

Comment:
As mentioned in one of previous comments, we will not need to check whether the nullifier is in the nullifier DB here.